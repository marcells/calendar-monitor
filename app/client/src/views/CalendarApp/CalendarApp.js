import React, { Component } from 'react';
import { connect } from 'react-redux'
import { loadCalendar } from '../../redux/actions';
import Calendar from '../../components/Calendar/Calendar';
import Upcoming from '../../components/Upcoming/Upcoming';
import './CalendarApp.css';

class CalendarApp extends Component {
  async componentDidMount() {
    this.props.dispatch(loadCalendar(this.props.calendar));

    this._timer = setInterval(() => this.props.dispatch(loadCalendar(this.props.calendar)), 10000);
  }

  componentWillUnmount() {
    if (this._timer) {
      clearInterval(this._timer);
    }
  }

  render() {
    return (
      <div className="Content">
        <div className="Calendars">
          { this.props.calendars.map(x => <Calendar key={this.props.calendar + x.date.year + x.date.month} calendar={x} />) }
        </div>

        <Upcoming events={this.props.upcoming} />
      </div>
    );
  }
}

function mapStateToProps(state, ownProps) {
  const { calendars, upcoming } = state.calendar;

  return {
    calendarId: ownProps.calendar,
    calendars,
    upcoming
  };
}

export default connect(mapStateToProps)(CalendarApp);
