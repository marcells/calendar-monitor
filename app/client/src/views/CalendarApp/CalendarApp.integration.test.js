import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import moxios from 'moxios';
import CalendarApp from './CalendarApp';

describe('CalendarApp', () => {
  describe('integration', () => {
    beforeEach(() => {
      moxios.install();
    });

    afterEach(() => {
      moxios.uninstall();
    });

    it('should render the calendars correctly', () => {
      const mockStore = configureStore([thunk]);
      const store = mockStore({
        calendar: {
          calendars: [
            {
              date: {
                month: 0,
                year: 2017
              },
              events: [{
                id: 'calendar1/1',
                title: 'Entry 1',
                from: new Date(2017, 0, 1, 17, 0, 0),
                to: new Date(2017, 0, 1, 19, 0, 0)
              },
              {
                id: 'calendar1/2',
                title: 'Entry 2',
                from: new Date(2017, 0, 18, 7, 0, 0),
                to: new Date(2017, 0, 18, 10, 0, 0)
              }]
            },
            {
              date: {
                month: 1,
                year: 2017
              },
              events: []
            }
          ],
          upcoming: [{
            id: 'calendar1/1',
            title: 'Entry 1',
            from: new Date(2017, 0, 1, 17, 0, 0),
            to: new Date(2017, 0, 1, 19, 0, 0)
          },
          {
            id: 'calendar1/2',
            title: 'Entry 2',
            from: new Date(2017, 0, 18, 7, 0, 0),
            to: new Date(2017, 0, 18, 10, 0, 0)
          }]
        },
        eventDetails: {
          event: { tags: [] }
        }
      });

      const mockedDispatch = jest.fn();

      const result = renderer.create(
        <Provider store={store}>
          <CalendarApp calendar='calendar1' dispatch={mockedDispatch} />
        </Provider>
      );

      expect(result).toMatchSnapshot();
    });
  });
});
