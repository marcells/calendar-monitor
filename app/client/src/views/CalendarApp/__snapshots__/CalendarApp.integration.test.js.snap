// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarApp integration should render the calendars correctly 1`] = `
<div
  className="Content"
>
  <div
    className="Calendars"
  >
    <div
      className="Calendar"
    >
      <div
        className="Calendar-header"
      >
        <div
          className="Calendar-header-month"
        >
          January
        </div>
        <div
          className="Calendar-header-weekdays"
        >
          <div>
            Sunday
          </div>
          <div>
            Monday
          </div>
          <div>
            Tuesday
          </div>
          <div>
            Wednesday
          </div>
          <div>
            Thursday
          </div>
          <div>
            Friday
          </div>
          <div>
            Saturday
          </div>
        </div>
      </div>
      <div
        className="Calendar-days"
      >
        <div
          className="Calendar-day Calendar-day-is-weekend "
          style={
            Object {
              "gridColumn": 1,
            }
          }
        >
          <div
            className="Calendar-day-column"
          >
            1
          </div>
          <div
            className="Calendar-day-events"
          >
            <div
              className="Calendar-day-event"
            >
              <span
                className="Calendar-day-event-from"
              >
                17:00
              </span>
              <span
                className="Calendar-day-event-title"
              >
                Entry 1
              </span>
            </div>
          </div>
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            2
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            3
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            4
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            5
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            6
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-weekend "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            7
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-weekend "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            8
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            9
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            10
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            11
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            12
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            13
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-weekend "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            14
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-weekend "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            15
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            16
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            17
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            18
          </div>
          <div
            className="Calendar-day-events"
          >
            <div
              className="Calendar-day-event"
            >
              <span
                className="Calendar-day-event-from"
              >
                07:00
              </span>
              <span
                className="Calendar-day-event-title"
              >
                Entry 2
              </span>
            </div>
          </div>
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            19
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            20
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-weekend "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            21
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-weekend "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            22
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            23
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            24
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            25
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            26
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            27
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-weekend "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            28
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-weekend "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            29
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            30
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            31
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
      </div>
    </div>
    <div
      className="Calendar"
    >
      <div
        className="Calendar-header"
      >
        <div
          className="Calendar-header-month"
        >
          February
        </div>
        <div
          className="Calendar-header-weekdays"
        >
          <div>
            Sunday
          </div>
          <div>
            Monday
          </div>
          <div>
            Tuesday
          </div>
          <div>
            Wednesday
          </div>
          <div>
            Thursday
          </div>
          <div>
            Friday
          </div>
          <div>
            Saturday
          </div>
        </div>
      </div>
      <div
        className="Calendar-days"
      >
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={
            Object {
              "gridColumn": 4,
            }
          }
        >
          <div
            className="Calendar-day-column"
          >
            1
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            2
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            3
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-weekend "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            4
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-weekend "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            5
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            6
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            7
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            8
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            9
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            10
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-weekend "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            11
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-weekend "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            12
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            13
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            14
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            15
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            16
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            17
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-weekend "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            18
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-weekend "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            19
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            20
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            21
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            22
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            23
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            24
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-weekend "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            25
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-weekend "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            26
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            27
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
        <div
          className="Calendar-day Calendar-day-is-workday "
          style={Object {}}
        >
          <div
            className="Calendar-day-column"
          >
            28
          </div>
          <div
            className="Calendar-day-events"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="Upcoming"
  >
    <div
      className="Upcoming-header"
    >
      Upcoming Events
    </div>
    <div>
      <div
        className="Upcoming-event"
      >
        <div
          className="Upcoming-event-title"
        >
          Entry 1
        </div>
        <div
          className="Upcoming-event-from"
        >
          01/01/2017
        </div>
      </div>
      <div
        className="Upcoming-event"
      >
        <div
          className="Upcoming-event-title"
        >
          Entry 2
        </div>
        <div
          className="Upcoming-event-from"
        >
          01/18/2017
        </div>
      </div>
    </div>
  </div>
</div>
`;
